src/fu-main.c

```diff
@@ -629,6 +630,11 @@ fu_main_install_with_helper (FuMainAuthHelper *helper_ref, GError **error)
 				continue;
 			}
 
+			/* if component should have an update message from CAB */
```

Comment:
I agree this needs doing, but it seems like the wrong place to do it. Maybe if we split up `fu_engine_set_release_from_appstream()` we could call into something shared like ` fu_engine_set_device_from_component()` that would also set things like `UpdateProtocol` too.

Reply:
`fu_engine_set_release_from_appstream` doesn't actually set the device though, only the release.  so that would mean maybe instead having install try to use `fu_engine_set_release_from_appstream` and a new helper function to copy things to the device